modules/mono/glue/GodotSharp/GodotSharp/Core/Bridge/ScriptManagerBridge.cs

```diff
@@ -448,6 +448,23 @@ internal static unsafe void GetOrCreateScriptBridgeForPath(godot_string* scriptP
             GetOrCreateScriptBridgeForType(scriptType, outScript);
         }
 
+        [UnmanagedCallersOnly]
+        internal static unsafe godot_bool GetExistingScriptBridgeForPath(godot_string* scriptPath, godot_ref* outScript)
+        {
+            string scriptPathStr = Marshaling.ConvertStringToManaged(*scriptPath);
+
+            if (_pathTypeBiMap.TryGetScriptType(scriptPathStr, out Type? scriptType))
+            {
+                if (_scriptTypeBiMap.TryGetScriptPtr(scriptType, out IntPtr scriptPtr))
+                {
+                    NativeFuncs.godotsharp_ref_new_from_ref_counted_ptr(out *outScript, scriptPtr);
+                    return godot_bool.True;
+                }
```

Comment:
This should probably use `_scriptTypeBiMap.ReadWriteLock` [as seen in `GetOrCreateScriptBridgeForType`](https://github.com/godotengine/godot/pull/81541/files#diff-597dfc729c32276fb0593ca6f8bf3b2e2e847ae52217a06f60c72452a2dda3dfR470-R481).
```suggestion
                lock (_scriptTypeBiMap.ReadWriteLock)
                {
                    if (_scriptTypeBiMap.TryGetScriptPtr(scriptType, out IntPtr scriptPtr))
                    {
                        NativeFuncs.godotsharp_ref_new_from_ref_counted_ptr(out *outScript, scriptPtr);
                        return godot_bool.True;
                    }
                }
```